src/main/java/org/terasology/changingBlocks/conditional/BlockCondition.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.changingBlocks.conditional;
+
+import org.terasology.math.Side;
+
+public class BlockCondition {
```

Comment:
They are part of the component, and thus should not provide much, if any, logic themselves. They mainly exist as classes just because I cannot otherwise figure out how to nest the required data within a prefab, where the component has a list of objects which each have named values.

Though, come to think of it, I'm not sure I actually have that working or not yet...

Reply:
It would be great to have it be a more modular mix-and-match, but since components only support variables, not methods, and Java doesn't support proper multiple inheritance...